### PR TITLE
Upgrade to Netty 4.1.31 and use individual dependencies

### PR DESCRIPTION
**Pull request by merlimat**
### Motivation

 * Upgrade to latest Netty version which brings in perf improvements and some new features (eg: https://github.com/netty/netty/pull/8267) 

 * Broke down the dependencies from `netty-all` into individual components, as discussed at https://github.com/apache/bookkeeper/pull/1755#discussion_r228449352

### CODE REVIEW

File: bookkeeper-dist/src/main/resources/LICENSE-server.bin.txt [hunk 1]
```diff
@@ -303,71 +317,63 @@ Apache Software License, Version 2.
 [36] Source available at https://github.com/apache/yetus/tree/rel/0.5.0
 
 ------------------------------------------------------------------------------------
-lib/io.netty-netty-all-4.1.22.Final.jar bundles some 3rd party dependencies
+lib/- lib/io.netty-netty-codec-4.1.31.Final.jar bundles some 3rd party dependencies
```
Comment (sijie) [inline bookkeeper-dist/src/main/resources/LICENSE-server.bin.txt]: this sounds like a replacement issue?

File: pom.xml [hunk 2]
```diff
@@ -371,7 +371,58 @@
...
+        <version>${netty.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>io.netty</groupId>
+        <artifactId>netty-codec-http2</artifactId>
+        <version>${netty.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>io.netty</groupId>
+        <artifactId>netty-codec-socks</artifactId>
```
Comment (eolivelli) [inline pom.xml]: Do we need this one?

File: pom.xml [hunk 3]
```diff
@@ -371,7 +371,58 @@
...
+        <version>${netty.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>io.netty</groupId>
+        <artifactId>netty-handler-proxy</artifactId>
```
Comment (eolivelli) [inline pom.xml]: Do we need this one?

File: pom.xml (hunk 2 above)
Comment (merlimat) [reply pom.xml]: This is already being pulled by other dependencies (eg: finagle, etc..). Here, I'm not adding a new dependency, just pinning all the artifacts to be on the same version.

File: pom.xml (hunk 3 above)
Comment (merlimat) [reply pom.xml]: same as above

File: bookkeeper-dist/src/main/resources/LICENSE-server.bin.txt (hunk 1 above)
Comment (merlimat) [reply bookkeeper-dist/src/main/resources/LICENSE-server.bin.txt]: Yes, will fix

**Comment by sijie** [comment]
run integration tests

File: bookkeeper-dist/src/main/resources/NOTICE-all.bin.txt [hunk 4]
```diff
@@ -45,7 +45,23 @@ License for the specific language governing permissions and limitations
...
+- lib/io.netty-netty-codec-dns-4.1.31.Final.jar
+- lib/io.netty-netty-codec-http-4.1.31.Final.jar
+- lib/io.netty-netty-codec-http2-4.1.31.Final.jar
+- lib/io.netty-netty-codec-socks-4.1.31.Final.jar
+- lib/io.netty-netty-common-4.1.31.Final.jar
+- lib/io.netty-netty-handler-4.1.31.Final.jar
+- lib/io.netty-netty-handler-proxy-4.1.31.Final.jar
+- lib/io.netty-netty-resolver-4.1.31.Final.jar
+- lib/io.netty-netty-resolver-dns-4.1.31.Final.jar
+- lib/io.netty-netty-tcnative-boringssl-static-2.0.19.Final.jar
```
[the remark this answers is not in the document]
Comment (eolivelli) [reply bookkeeper-dist/src/main/resources/NOTICE-all.bin.txt]: That one (the static ssl lib) is not part of -all

**Comment by merlimat** [comment]
run integration tests

**Comment by merlimat** [comment]
@eolivelli can you please check again?

**Comment by merlimat** [comment]
@eolivelli I think it's good to keep it for 4.9.0. The dependency change (from all to individuals) might surprise users in a patch release.

Also, there's no specific bug that was fixed in Netty-4.1.31 that we need to make available.

**Comment by eolivelli** [comment]
@merlimat okay.
btw (Probably) I will be running soon bk 4.8.1 + latest netty version.

**Pull request merged**